### PR TITLE
Fix 271

### LINKED ISSUE #271: Temporal subsetting fails checks in start_date or end_date not within dataset time bounds
* xclim version: '0.10.10-beta'
* Python version: 3.6
* Operating System: Linux

### Description
if da.time.sel(time=start_date).min() > da.time.sel(time=end_date).max():
Temporal subset fails if end_date is not in selection.

See #190

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated

### After merging to master and before closing the branch:
- [ ] bumpversion (minor / major / patch) has been called on `master` branch
- [ ] Tags have been pushed

* **What kind of change does this PR introduce?** 

* **Does this PR introduce a breaking change?** 

* **Other information**:

This PR fixes #271

### CODE REVIEW

File: xclim/subset.py [hunk 1]
```diff
@@ -115,7 +160,7 @@ def subset_bbox(da, lon_bnds=None, lat_bnds=None, start_date=None, end_date=None
 
     Returns
     -------
-    xarray.DataArray or xarray.DataSet
+    Union[xarray.DataArray, xarray.DataSet]
       subsetted data array or dataset
```
Comment (huard) [inline xclim/subset.py]: Capitalize

File: xclim/subset.py [hunk 2]
```diff
@@ -43,14 +45,58 @@ def func_checker(*args, **kwargs):
...
+
+        return func(*args, **kwargs)
+
+    return func_checker
+
+
 def check_lons(func):
     def func_checker(*args, **kwargs):
         """
         A decorator to reformat user-specified lon values based on the lon dimensions of a supplied xarray
```
Comment (huard) [inline xclim/subset.py]: I think the docstring could be more explicit about what is done in this decorator.

Comment (Zeitsperre) [reply xclim/subset.py]: Done.

File: xclim/subset.py (hunk 1 above)
Comment (Zeitsperre) [reply xclim/subset.py]: Done.

**Pull request merged**